paging/common/src/main/kotlin/androidx/paging/PageFetcher.kt

```diff
@@ -143,18 +139,16 @@ internal class PageFetcher<Key : Any, Value : Any>(
             launch {
                 var prev = LoadStates.IDLE
                 accessor.state.collect {
-                    if (prev.refresh != it.refresh) {
+                    if (prev != it) {
+                        val prevCombinedLoadStates = loadStates.snapshot()
+
                         loadStates.set(REFRESH, true, it.refresh)
-                        dispatchIfValid(REFRESH, it.refresh)
-                    }
-                    if (prev.prepend != it.prepend) {
                         loadStates.set(PREPEND, true, it.prepend)
-                        dispatchIfValid(PREPEND, it.prepend)
-                    }
-                    if (prev.append != it.append) {
                         loadStates.set(APPEND, true, it.append)
-                        dispatchIfValid(APPEND, it.append)
+
+                        dispatchIfValid(prevCombinedLoadStates)
```

Comment:
@dlam Would we still want to check if the load states are valid to be dispatched here just to be thorough, or should we just send them without checking? Since as far as I understand, if there are any changes between the `prev` load states and the new states from the accessor, they'll be remote changes and will all be valid to be dispatched?

Reply:
These checks have always been around just to be thorough because it's very easy to mess up, so we should keep it.